hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/QueryHandler.java

```diff
@@ -15,19 +15,52 @@
  */
 package com.hedera.node.app.spi.workflows;
 
-import com.hederahashgraph.api.proto.java.TransactionBody;
+import com.hederahashgraph.api.proto.java.Query;
+import com.hederahashgraph.api.proto.java.QueryHeader;
+import com.hederahashgraph.api.proto.java.Response;
+import com.hederahashgraph.api.proto.java.ResponseHeader;
+import com.hederahashgraph.api.proto.java.ResponseType;
 import edu.umd.cs.findbugs.annotations.NonNull;
 
 /** A {@code QueryHandler} contains all methods for the different stages of a single query. */
 public interface QueryHandler {
 
     /**
-     * This method is called during the query-workflow. It does pre-checks that are specific to the
-     * operation.
+     * Extract the {@link QueryHeader} from a given {@link Query}
      *
-     * @param txBody the {@link TransactionBody} that needs to be validated
-     * @throws NullPointerException if {@code txBody} is {@code null}
-     * @throws PreCheckException if validation fails
+     * @param query the {@link Query} that contains the header
+     * @return the {@link QueryHeader} that was specified in the query
+     * @throws NullPointerException if {@code query} is {@code null}
      */
-    void preCheck(@NonNull TransactionBody txBody) throws PreCheckException;
+    QueryHeader extractHeader(@NonNull Query query);
+
+    /**
+     * Creates an empty {@link Response} with a provided header. This is typically used, if an error
+     * occurred.
```

Comment:
Probably mention that the ResponseHeader will have error code in it.

Reply:
Added